backend/src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -29,12 +29,22 @@
  * - if a user is logged in, it executes corresponding abstract method get or post
  */
 public abstract class AuthenticationServlet extends HttpServlet {
-  private static final int AUTHENTICATION_ERROR_CODE = 401;
+  public static final int AUTHENTICATION_ERROR_CODE = 401;
+  protected static UserService mUserService;
+
+  public AuthenticationServlet() {
+    mUserService = UserServiceFactory.getUserService();
+  }
+
+  // For testing, UserService is mocked
```

Comment:
can you instead make a function called setUserServiceForTesting() and annotate it with @VisibleForTesting?